source/h2_decoder.c

```diff
@@ -245,6 +248,12 @@ struct aws_h2_decoder {
          * A malformed header-block is not a connection error, it's a Stream Error (RFC-7540 5.4.2).
          * We continue decoding and report that it's malformed in on_headers_end(). */
         bool malformed;
+
+        /* Buffer up cookie header fields to concatenate separate ones */
+        struct aws_byte_buf cookies;
+        /* Compression type for the concatenated cookie header. Let's say if any separate one is not cached, the last
+         * one will not be cached */
+        enum aws_http_header_compression cookie_header_compression_type;
```

Comment:
/* If separate cookie fields have different compression types, the concatenated cookie uses the strictest type. */